content/documentation/4-batch-developer-guides/2-batch/3-data-flow-simple-task.md

```diff
@@ -1,32 +1,104 @@
 
 ### Application Registration
 
-how to register apps using the CLI and/or (?) UI
+The Data Flow Dashboard will land on the Application Registration view where we will register the sample task.
+
+![Add an application](images/SCDF-add-applications.png)
+
+#### Application Registration Concepts
+
+Applications in Data Flow are registered as named resources so that they may be referenced when using the Data Flow DSL to configure and compose tasks.
+Registration associates a logical application name and type with a physical resource, given by a URI.
+The URI conforms to a [schema](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#spring-cloud-dataflow-register-stream-apps) and may represent a Maven artifact, a Docker image, or an actual `http(s)` or `file` URL.
+Data Flow defines a some logical application types which indicate its role as a streaming component, a task, or a standalone application.
+Spring Cloud Task applications, as you might guess, are always registered as a `task` type.
+
+##### Maven Artifacts
+
+The URI for a Maven artifact is of the form `maven://<groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>`.
+
+[[note | Application Metadata]]
+|The `classifier` and `extension` specifiers are optional for the Maven URI schema and normally not used for application artifacts.
+|The pre-built [stream](http://cloud.spring.io/spring-cloud-stream-app-starters/) and [task/batch](http://cloud.spring.io/spring-cloud-task-app-starters/) applications use a custom Maven plugin to extract the application metadata, including application property descriptors, into a separate jar file.
+|In this case, in addition to the executable application artifact, we register an additional metadata artifact using the `metadata` classifier.
+
+The maven URI for the sample app is:
+
+```
+maven://io.spring:billsetuptask:1.0.0.BUILD-SNAPSHOT
+```
+
+The `maven:` protocol specifies a Maven artifact which is resolved using the remote and local Maven repositories configured for the Data Flow Server.
+
+##### Docker Images
+
+The URI for a Docker Image is of the form `docker:<docker-image-path>/<imageName>:<version>` and is resolved using the Docker registry configured for the Data Flow task platform and image pull policy.
+
+The Docker URI for the sample app is:
+
+```
+docker:springcloudtask/billsetuptask:1.0.0.BUILD-SNAPSHOT
+```
+
+#### Registering an Application
+
+To register an application, select `Add Applications` and `Register one or more applications`. Fill in the form, as shown, and hit `Register the application(s)`.
+
+![Register the billsetup Task app](images/SCDF-register-task-app-maven.png)
+
```

Comment:
Needs an extra blank line here or the image will not render.

Reply:
Hmmm, it renders fine for me.